scripts/polyrith_sage_helper.py

```diff
@@ -15,20 +17,46 @@ def power_to_string(var: int, pow: int) -> str:
         return var_s
     return mk_app("poly.pow", var_s, str(pow))
 
-def sum_to_string(terms: list[str]) -> str:
-    terms = iter(terms)
+@dataclass
+class MonomForm:
+    pos_form: str
+    neg_form: Optional[str] = None
+
+def sum_to_string_aux(old: str, rest: list[MonomForm]) -> str:
+    try:
+        nxt = next(rest)
+    except StopIteration:
+        return old
+    s = mk_app("poly.sub" if nxt.neg_form is not None else "poly.add", old, nxt.pos_form)
+    return sum_to_string_aux(s, rest)
+
+def sum_to_string(terms: list[MonomForm]) -> str:
```

Comment:
```suggestion
def sum_to_string(terms: Iterator[MonomForm]) -> str:
```
(`from typing import Iterator`)
otherwise `next` would be an error below